sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -125,11 +129,12 @@ int main(int argc, char **argv) {
     const auto &Devices = Platform.get_devices();
     if (verbose)
       std::cout << "    Devices  : " << Devices.size() << std::endl;
-    uint32_t DeviceNum = 0;
     for (const auto &Device : Devices) {
-      ++DeviceNum;
       if (verbose)
         std::cout << "        Device [#" << DeviceNum << "]:" << std::endl;
+      else
```

Comment:
what is this printing an why?

Reply:
We need to print the device number from sycl-ls so that users can use the device number when setting SYCL_DEVICE_FILTER.
Here is current output of sycl-ls before my PR:
```
GPU : Intel(R) OpenCL HD Graphics 3.0 [21.04.18912]
GPU : Intel(R) Level-Zero 1.0 [1.0.18912]
HOST: SYCL host platform 1.2 [1.2]
```
Here is the output after my PR:
```
1. GPU : Intel(R) OpenCL HD Graphics 3.0 [21.04.18912]
2. GPU : Intel(R) Level-Zero 1.0 [1.0.18912]
3. HOST: SYCL host platform 1.2 [1.2]
```

Reply:
Now, the sycl-ls output looks like this.
```
0. ACC : Intel(R) FPGA Emulation Platform for OpenCL(TM) 1.2 [2020.11.11.0.04_160000]
1. GPU : Intel(R) OpenCL HD Graphics 3.0 [21.04.18912]
2. CPU : Intel(R) OpenCL 2.1 [2020.11.11.0.04_160000]
3. GPU : Intel(R) Level-Zero 1.0 [1.0.18912]
4. HOST: SYCL host platform 1.2 [1.2]
```

Reply:
Just came to me that may be we should have "0" always be the host?

Reply:
SYCL 2020 removes the host device. Therefore I don't think it makes sense to use it as 0.

Reply:
Remember that SYCL 2020 no longer has a special "host" device.  I presume we will continue to have a device that is implemented on the host CPU, but it will probably be classified as a "cpu" device once we are SYCL 2020 conformant.  I don't think we want customer code to start assuming that device 0 is the host device, since that will probably not be true in the long term.

Reply:
Ok, fair point.